### ISSUE TITLE
__module__ of classes imported from __init__.py is confusing for sphinx

### ISSUE THREAD

**Issue by cblegare** [ets-labs/python-dependency-injector#147]
Sphinx (using [intersphinx](http://www.sphinx-doc.org/en/stable/ext/intersphinx.html)) can't find some classes from your objects.inv, resulting in 

```text
WARNING: py:class reference target not found: dependency_injector.containers.declarative.DeclarativeContainer
```

because it is referenced as `dependency_injector.containers.DeclarativeContainer` and not `dependency_injector.containers.declarative.DeclarativeContainer` in the [objects.inv](http://python-dependency-injector.ets-labs.org/objects.inv)

This can be worked around by assigning the `__module__` property of classes imported in any `__init__.py`.  See also http://stackoverflow.com/a/24787344

**Comment by rmk135** [comment]
Hi @cblegare,

True story, needs to be fixed. I will update this issue, when it's done.

Thanks,
Roman

**Comment by rmk135** [comment]
Ok, so, it's been fixed for containers in 3.3.4 that's already on PyPi (https://pypi.python.org/pypi/dependency_injector/3.3.4). Now it should be alright:

```python
>>> import dependency_injector.containers
>>> dependency_injector.containers.DeclarativeContainer
<class 'dependency_injector.containers.DeclarativeContainer'>
>>> dependency_injector.containers.DeclarativeContainer.__module__
'dependency_injector.containers'
```

At the same time there is no chances currently to apply the same for providers while they are written in C (Cython) and Python says that they couldn't be monkeypatched:

```python
>>> import dependency_injector.providers
>>> dependency_injector.providers.Factory
<type 'dependency_injector.providers.factories.Factory'>
>>> dependency_injector.providers.Factory.__module__ = 'dependency_injector.providers'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't set attributes of built-in/extension type 'dependency_injector.providers.factories.Factory'
```

As for now, I'm not sure what I can do, but there should be a solution...

**Comment by rmk135** [comment]
Fixed completely in 3.3.5.

I've consolidated all providers in single module. Same was done for containers. Now containers and providers are truly in their own modules.

```python
>>> import dependency_injector.containers
>>> import dependency_injector.providers
>>>
>>>
>>> dependency_injector.containers.DeclarativeContainer
<class 'dependency_injector.containers.DeclarativeContainer'>
>>> dependency_injector.providers.Factory
<type 'dependency_injector.providers.Factory'>
```

Closing this issue,
Thanks once more for your contribution,
Roman

**Comment by cblegare** [comment]
Thank you!
